File: application/src/main/java/com/study/platform/domain/post/usecase/CreateStudyPostUseCase.java
package com.study.platform.domain.post.usecase;

import com.study.platform.domain.post.dto.request.StudyPostCreateRequest;
import com.study.platform.domain.post.dto.response.StudyPostResponse;

import java.util.UUID;

public interface CreateStudyPostUseCase {
    StudyPostResponse execute(UUID userId, StudyPostCreateRequest request);
}

File: application/src/main/kotlin/com/study/platform/domain/post/usecase/CreateStudyPostUseCase.kt
package com.study.platform.domain.post.usecase

import com.study.platform.domain.post.dto.request.StudyPostCreateRequest
import com.study.platform.domain.post.dto.response.StudyPostResponse
import java.util.UUID

interface CreateStudyPostUseCase {
    fun execute(userId: UUID, request: StudyPostCreateRequest): StudyPostResponse
}
